transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -842,10 +843,8 @@ public ChannelFuture writeAndFlush(Object msg) {
     }
 
     private static void notifyOutboundHandlerException(Throwable cause, ChannelPromise promise) {
-        if (!promise.tryFailure(cause) && !(promise instanceof VoidChannelPromise)) {
-            if (logger.isWarnEnabled()) {
-                logger.warn("Failed to fail the promise because it's done already: {}", promise, cause);
-            }
+        if (!(promise instanceof VoidChannelPromise)) {
```

Comment:
I think for 4.0 we can only do it like I did here :( For 4.1 we can do it with `isVoid(...)` directly in the `PromiseNotificationUtil` methods. WDYT ?